### PR TITLE
Use official github icons

### PR DESCRIPTION
**Pull request by AugustasV**
#### Summary

Tried to change icons from SVG to github ones. Not sure how to test it.

#### Ticket Link

  Fixes https://github.com/mattermost/mattermost-plugin-github/issues/268

### CODE REVIEW

File: webapp/src/images/icons/changes_requested.jsx [hunk 1]
```diff
@@ -2,22 +2,17 @@
...
-                    fillRule='evenodd'
-                    d='M0 1a1 1 0 011-1h14a1 1 0 011 1v10a1 1 0 01-1 1H7.5L4 15.5V12H1a1 1 0 01-1-1V1zm1 0v10h4v2l2-2h8V1H1zm7.5 3h2v1h-2v2h-1V5h-2V4h2V2h1v2zm2 5h-5V8h5v1z'
-                />
-            </svg>
-        );
-    }
-}
+export default function OcticonsList() {
+  return (
+    <ul>
```
Comment (larkox) [inline webapp/src/images/icons/changes_requested.jsx:9]: Not sure why are you getting a list here. Cannot this be done just by something like:
```
return (
    <Octicon
        icon={iconType}
        size='small'
        verticalAlign='middle'
    />
);
```

Comment (AugustasV) [reply webapp/src/images/icons/changes_requested.jsx]: No idea. Removed list, just added icons.

**Comment by larkox** [comment]
@AugustasV Have you been able to check Hanzei's comments?

**Comment by AugustasV** [comment]
@larkox yes, tried to run `make check-style` on my machine but unsuccessfully. But right now as I could see there is no more linter errors.

**Comment by AugustasV** [comment]
> Clicking on the icons on the RHS causes the webapp to crash for me.
> Screenshot from 2020-06-16 21-35-28
> 
> Could you please look into this?

@larkox how should I debug this? I don't think Circleci would help me now.

**Comment by larkox** [comment]
@AugustasV According to [octicon documentation](https://primer.style/octicons/packages/react) the size parameter only accept strings, and those string being "small", "medium" or "large". We should use "small" in this context.

File: webapp/src/images/icons/changes_requested.jsx [hunk 2]
```diff
@@ -2,22 +2,8 @@
...
-                    fillRule='evenodd'
-                    d='M0 1a1 1 0 011-1h14a1 1 0 011 1v10a1 1 0 01-1 1H7.5L4 15.5V12H1a1 1 0 01-1-1V1zm1 0v10h4v2l2-2h8V1H1zm7.5 3h2v1h-2v2h-1V5h-2V4h2V2h1v2zm2 5h-5V8h5v1z'
-                />
-            </svg>
-        );
-    }
-}
+export default () => (
+    <GitPullRequestIcon size='small'/>
+);
```
Comment (hanzei) [inline webapp/src/images/icons/changes_requested.jsx:9]: Nit pick: Please add an empty line here.

File: webapp/src/images/icons/cross.jsx [hunk 3]
```diff
@@ -2,22 +2,8 @@
...
-                    fillRule='evenodd'
-                    d='M7.48 8l3.75 3.75-1.48 1.48L6 9.48l-3.75 3.75-1.48-1.48L4.52 8 .77 4.25l1.48-1.48L6 6.52l3.75-3.75 1.48 1.48L7.48 8z'
-                />
-            </svg>
-        );
-    }
-}
+export default () => (
+    <XIcon size='small'/>
+);
```
Comment (hanzei) [inline webapp/src/images/icons/cross.jsx:9]: Same here

File: webapp/src/images/icons/dot.jsx [hunk 4]
```diff
@@ -2,22 +2,8 @@
...
-                    fillRule='evenodd'
-                    d='M0 8c0-2.2 1.8-4 4-4s4 1.8 4 4-1.8 4-4 4-4-1.8-4-4z'
-                />
-            </svg>
-        );
-    }
-}
+export default () => (
+    <DotIcon size='small'/>
+);
```
Comment (hanzei) [inline webapp/src/images/icons/dot.jsx:9]: And here.

**Comment by hanzei** [comment]
@DHaussermann This might be a good candidate for dev testing.

**Comment by DHaussermann** [comment]
@larkox or @hanzei, If you have a bit of time, could one of you peer-test this PR using the  peer testing process documented here https://mattermost.atlassian.net/wiki/spaces/INT/pages/619937892/Peer+Testing+Process

**Comment by AugustasV** [comment]
@hanzei so I should rewrite my app to another location? I am confused. It was working before, you made changes and now it doesn't work or what happened?

**Comment by hanzei** [comment]
Sorry for the confusion. Let me see, if I can clarify. 

>  It was working before, you made changes and now it doesn't work or what happened?

I don't think my changes broke that. I'm encountering exactly the same issue with 71615f756484f8b04a1499ea2f60f942a4405f5a and 9c00df7d2342b2cec363ef2953505b89fee851dd.

> so I should rewrite my app to another location?

My thinking was, that the exported types in `webapp/src/images/icons/*` are only used in `webapp/src/components/sidebar_right/github_items.jsx`. Hence moving your code into that file might a) fix the console error above and  b) make the code cleaner.

It's just a suggestion.

Can you reproduce the console error and crash on your machine?

**Comment by AugustasV** [comment]
@hanzei I would like to try locally, but got this error then wrote make deploy ([was following development section ](https://github.com/mattermost/mattermost-plugin-github#development) )

**Comment by hanzei** [comment]
@AugustasV Did you setup your local server following the steps at https://developers.mattermost.com/contribute/server/developer-setup/?

**Comment by AugustasV** [comment]
@hanzei thanks, I managed to set up local environment, Register an OAuth application in GitHub and added details to mattermost environment. How I should test those icons? I didn't saw any errors yet.

**Comment by hanzei** [comment]
@AugustasV First you have to connect your GitHub account using `/github connect`. Once that is set up you can access the RHS by clicking on e.g. the pull request icon on the left hand side.

**Comment by AugustasV** [comment]
@hanzei I setup it locally, but couldn't reproduce error - everything works as expected.

**Comment by hanzei** [comment]
@larkox I'm wondering if you have any ideas on how to resolve the issue pointed out above. I did give it a try but didn't found a solution.

**Comment by larkox** [comment]
It is only a hunch, but what about naming the functions? Instead of:
```
export default () => (
    <GitPullRequestIcon size='small'/>
);
```
Something like:
```
export default function ChangesRequestedIcon() {
    return ( <GitPullRequestIcon size='small'/>);
}
```

**Review by hanzei — CHANGES_REQUESTED** [review]
Unfortunately, I'm still running into the same issue as in https://github.com/mattermost/mattermost-plugin-github/pull/293#pullrequestreview-462379034

cc @larkox

**Comment by larkox** [comment]
After digging deeper into this, I found the problem.
The current imports are like:
```
import GitPullRequestIcon from '@primer/octicons-react';
```
That means that we are importing the defult export of octicons-react (the component Octicon) with the name GitPullRequestIcon. Instead we should do:
```
import {GitPullRequestIcon} from '@primer/octicons-react';
```
That way we just import the GitPullRequestIcon export.

There might be another small problem regarding version. The version installed is 9.x, and on that version, most of the icons names do not have the suffix of "Icon". That is from version 10.x. Therefore we have two choices:
- Bump the version of octicon and fix other occurrences in the code
- Use the naming of 9.x

Also, they way the coloring is done at the moment will not work, so that should also be fixed.

**Comment by hanzei** [comment]
@AugustasV I've setup a cloud test server for you. You can use it to test you changes yourself, which might help with debugging the above failure.

**Review by hanzei — CHANGES_REQUESTED** [review]
I'm still getting an error message locally. It's a different one than the last time. 

Screenshot from 2020-09-04 00-29-48

**Comment by hanzei** [comment]
@AugustasV I've setup a cloud test server for you. You can use it to test you changes yourself, which might help with debugging the above failure.

**Comment by larkox** [comment]
Probably the problem is with the places in the code where octicons was already being used in the code. Probably you have to use the proper names for the icons.

**Comment by jasonblais** [comment]
@hanzei / @larkox is this ready for a review?

File: webapp/src/images/icons/sign.jsx [hunk 5]
```diff
@@ -2,20 +2,8 @@
...
-            >
-                <path
-                    d='M496 64H128V16c0-8.8-7.2-16-16-16H80c-8.8 0-16 7.2-16 16v48H16C7.2 64 0 71.2 0 80v32c0 8.8 7.2 16 16 16h48v368c0 8.8 7.2 16 16 16h32c8.8 0 16-7.2 16-16V128h368c8.8 0 16-7.2 16-16V80c0-8.8-7.2-16-16-16zM160 384h320V160H160v224z'
-                />
-            </svg>
-        );
-    }
-}
+export default () => (
+    <SignInIcon size='small'/>
```
Comment (hanzei) [inline webapp/src/images/icons/sign.jsx:8]: I think you want `MilestoneIcon` instead 
Screenshot from 2020-10-01 14-35-43

File: webapp/src/images/icons/dot.jsx [hunk 6]
```diff
@@ -2,22 +2,8 @@
...
-                <path
-                    fillRule='evenodd'
-                    d='M0 8c0-2.2 1.8-4 4-4s4 1.8 4 4-1.8 4-4 4-4-1.8-4-4z'
-                />
-            </svg>
-        );
-    }
-}
+export default () => (
+    <DotIcon size='small'/>
```
Comment (hanzei) [inline webapp/src/images/icons/dot.jsx:8]: This should be https://primer.style/octicons/dot-fill-16

Screenshot from 2020-10-01 14-36-32

File: webapp/src/images/icons/changes_requested.jsx [hunk 7]
```diff
@@ -2,22 +2,8 @@
...
-                <path
-                    fillRule='evenodd'
-                    d='M0 1a1 1 0 011-1h14a1 1 0 011 1v10a1 1 0 01-1 1H7.5L4 15.5V12H1a1 1 0 01-1-1V1zm1 0v10h4v2l2-2h8V1H1zm7.5 3h2v1h-2v2h-1V5h-2V4h2V2h1v2zm2 5h-5V8h5v1z'
-                />
-            </svg>
-        );
-    }
-}
+export default () => (
+    <GitPullRequestIcon size='small'/>
```
Comment (hanzei) [inline webapp/src/images/icons/changes_requested.jsx:8]: Is that really the right icon? Could you please post a screenshot?

**Review by hanzei — CHANGES_REQUESTED** [review]
Screenshot from 2020-10-01 14-38-34
The styling on the icons is gone. Could you please take a look why this is happening?

File: webapp/src/images/icons/changes_requested.jsx (hunk 7 above)
Comment (AugustasV) [reply webapp/src/images/icons/changes_requested.jsx:8]: https://primer.style/octicons/git-pull-request-16

Comment (hanzei) [reply webapp/src/images/icons/changes_requested.jsx:8]: That doesn't seem similar to the previous icon. Is there another that can be used?

Comment (AugustasV) [reply webapp/src/images/icons/changes_requested.jsx:8]: What about this one? That look familiar https://primer.style/octicons/git-branch-24

Comment (hanzei) [reply webapp/src/images/icons/changes_requested.jsx:8]: That looks reasonable. https://primer.style/octicons/diff-16 is more simplistic. I'm 0/5.

@asaadmahmood What do you think?

Comment (asaadmahmood) [reply webapp/src/images/icons/changes_requested.jsx:8]: @hanzei Fine with https://primer.style/octicons/file-diff-24

File: webapp/package.json [hunk 8]
```diff
@@ -64,8 +64,8 @@
     "webpack-cli": "3.3.12"
   },
   "dependencies": {
-    "@primer/octicons-react": "9.6.0",
-    "core-js": "3.6.5",
+    "@primer/octicons-react": "^10.1.0",
+    "core-js": "3.1.4",
```
Comment (hanzei) [inline webapp/package.json:68]: Why does `core-js` need go get downgraded?

**Review by hanzei — CHANGES_REQUESTED** [review]
The icons don't get proper styling. Could you please check why this is happening? 
 
Screenshot from 2020-10-07 13-32-30

File: webapp/package.json (hunk 8 above)
Comment (AugustasV) [reply webapp/package.json]: @hanzei Strange I don't know how it happened. Made this change back.

**Comment by asaadmahmood** [comment]
@AugustasV Can I get an updated screenshot in light and dark mode?

**Comment by asaadmahmood** [comment]
@AugustasV Can I get an updated screenshot in light and dark mode?

**Comment by AugustasV** [comment]
> @AugustasV Can I get an updated screenshot in light and dark mode?

@hanzei could you do that? not sure how to run this system on my pc, and right now I am quite busy.

**Review by asaadmahmood — APPROVED** [review]

**Comment by jfrerich** [comment]
@hanzei, reminder for a final review.  Thanks for all the feedback and discussion!

**Review by hanzei — CHANGES_REQUESTED** [review]
I'm still seeing the styling issue from https://github.com/mattermost/mattermost-plugin-github/pull/293#pullrequestreview-503782493.

Could you please check if the css needs to get adjusted?

**Comment by hanzei** [comment]
I talked with @AugustasV offline and resolved the CSS issue for him.

@DHaussermann This is ready for review.

**Review by hanzei — APPROVED** [review]

**Comment by jfrerich** [comment]
@DHaussermann bump for review.  @AugustasV, QA has been extremely busy lately, but will get to your PR soon!  Thanks for your patience

**Comment by DHaussermann** [comment]
Hi @AugustasV sorry for the delay in testing this. 

Thanks for the PR! It looks like the icon was updated successfully. However, when I click the icons in side bar, I see a react error rather than seeing the RHS open. 
`react.production.min.js:20 Uncaught Error: Minified React error #143; visit https://reactjs.org/docs/error-decoder.html?invariant=143 for the full message or use the non-minified dev environment for full errors and additional helpful warnings.`

I don't see this on the same server when I have the master branch deployed. Could this be related to this PR?
JS-Error

**Comment by hanzei** [comment]
@DHaussermann That issue was caused by merging #376 into this branch. I've fixed it. PTAL.

**Review by DHaussermann — APPROVED** [review]
Tested and passed
- Icon visible looks correct 
- Checked all areas in the UI and the marketplace
- Tested on Chrome and desktop app
- Checked on light and dark themes
- Unrelated issue with redux is now resolved (thanks @hanzei )

LGTM!

Thanks AugustasV for this enhancement!

**Pull request merged**